web/interceptor/access_control_test.go

```diff
@@ -0,0 +1,417 @@
+package interceptor_test
+
+import (
+	"context"
+	"errors"
+	"net"
+	"testing"
+
+	"github.com/quickfeed/quickfeed/ci"
+	"github.com/quickfeed/quickfeed/internal/qtest"
+	"github.com/quickfeed/quickfeed/qf"
+	"github.com/quickfeed/quickfeed/qlog"
+	"github.com/quickfeed/quickfeed/scm"
+	"github.com/quickfeed/quickfeed/web"
+	"github.com/quickfeed/quickfeed/web/auth"
+	"github.com/quickfeed/quickfeed/web/interceptor"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+const (
+	BufSize = 1024 * 1024
+)
+
+type accessTests []struct {
+	name     string
+	ctx      context.Context
+	userID   uint64
+	courseID uint64
+	groupID  uint64
+	access   bool
```

Comment:
Ref my comment regarding the `verifyAccess` func at the bottom of the file...
Would it not be better to replace this `access` field with a:
```go
wantErr error
```
You can simply define alias var something like this:
```go
var deniedErr = interceptor.ErrAccessDenied
```
that you can use in the struct tables to avoid long vars.

Reply:
Well, `wantErr` will only have two possible values: `ErrAccessDenied` and any other error (including nil). Is it really better than a Boolean?

Reply:
Not sure. Let's leave it as is for now.